src/sparseml/modifiers/utils/layer_compressors.py

```diff
@@ -14,37 +14,44 @@
 
 import inspect
 import logging
+from abc import ABC
 from typing import Dict, List
 
 import torch
 from torch.nn import Module
 
-from sparseml.modifiers.utils.gpt_helpers import SparseGPT
+from sparseml.modifiers.utils.gpts import GPT, SparseGPT, WandaGPT
 from sparseml.pytorch.utils.helpers import get_dependency_order
 from sparseml.utils.pytorch.module import get_prunable_layers
 
 
 _LOGGER = logging.getLogger(__name__)
 
 
-class LayerCompressor:
+class LayerCompressor(ABC):
     """
-    Runs the SparseGPT algorithm on a single layer using calibration data inputs
+    Defines the contract to run GPT algorithms on a
+    single layer using calibration data inputs
 
-    Lifecycle:
+    Example Lifecycle:
         - compress
             - pre_compress_parallel (optional)
             - add_batch
             - fasterprune
             - post_compress
 
+    Note: inheriting classes must define the gpt_class attribute,
+    and implement the invoke_fasterprune, and compress methods.
+
     :param model: model containing the layer we are running compression on
     :param layer: layer to run compression on
     :param layer_index: index of layer in the model
     :param inputs: calibration data to pass through the layer
     :param args: additional keyword arguments
     """
 
+    gpt_class: GPT
```

Comment:
See my note in the other PR, I think it makes more sense to call this terminal_module_processor or something